.github/scripts/build_universal_macos.sh

```diff
@@ -1,7 +1,46 @@
 #!/bin/bash -e
+shopt -s extglob
+
+readonly temp_dir="$(mktemp -d)"
+
+# macOS Monterey, v12, is the earliest macOS version with bottles in brew.
```

Comment:
The reason we're here is that these versions are not static; homebrew removes bottles for earlier versions if they bump their support matrix.

We don't want a hardcoded dependency on homebrew anymore that can break suddenly. See https://github.com/neovim/neovim/pull/25534 for previous discussion.